### PR TITLE
Add toString on RoleInterface

### PR DESCRIPTION
**Pull request by Orkin**
When we need to test if identity has one or more role

### CODE REVIEW

File: src/Role/RoleInterface.php [hunk 1]
```diff
@@ -30,4 +30,11 @@ public function getName();
      * @return bool
      */
     public function hasPermission($permission);
+    
+    /**
+     * Transform role to string
+     *
+     * @return string
+     */
+    public function __toString();
```
Comment (Ocramius) [inline src/Role/RoleInterface.php]: This is a major BC break

Comment (Orkin) [inline src/Role/RoleInterface.php]: I can remove this from the interface ;)

**Comment by bakura10** [comment]
I'd say this is an incorrect usage of it. What you want to test if it is has the permission, not the role.

The fact that the role has the permission is an implementation concern.

But if it can helps I'm not against this change (but indeed not in the interface).

**Comment by Orkin** [comment]
Yes I agree but sometime you just want to test if (most of the time on HierarchicalRole case) a use has one role.

**Comment by bakura10** [comment]
Actually after reading the code again, what's the difference with the "getName" method?

**Comment by Orkin** [comment]
In the example :

``` php
$identityRoles = $this->flattenRoles($identityRoles); // Generator
return count(array_intersect($roleNames, iterator_to_array($identityRoles))) > 0; //array_intersect cast each identityRole to string
```

So implement a __toString function in Role is useful but I can do it like this 👍 

``` php
$identityRoles = $this->flattenRoles($identityRoles);

$identityRolesString = [];
foreach ($identityRoles as $identityRole) {
    if ($identityRole instanceof RoleInterface) {
        $identityRole = $identityRole->getName();
    }
    $identityRolesString[] = (string) $identityRole;
}

return count(array_intersect($roleNames, $identityRolesString)) > 0;
```

With this I don't need __toString() function

**Comment by bakura10** [comment]
What about this:

``` php
$identityRoles = $this->flattenRoles($identityRoles); // Generator
return count(array_intersect($roleNames, array_map(function($item) {
  return $item instanceof RoleInterface ? $item->getName() : $item;
}, $identityRoles)) > 0; //array_intersect cast each identityRole to string
```

**Comment by Orkin** [comment]
Nop because $identityRoles is a Generator so we have to transform it to array with iterator_to_array

``` php
return count(array_intersect($roleNames, array_map(function ($item) {
    return $item instanceof RoleInterface ? $item->getName() : $item;
}, iterator_to_array($identityRoles)))) > 0;
```

**Comment by bakura10** [comment]
@danizord is this ok to merge?

**Comment by danizord** [comment]
Ok for me :+1: but checking for roles is definitely not a recommended practice in ZfrRbac, since roles are made to be dynamic (admins can create/update/delete roles, so if you hardcode it, it would stop working).

**Pull request closed without merging**